core/connector/src/main/java/org/exoplatform/wcm/connector/collaboration/DocumentEditorsRESTService.java

```diff
@@ -303,9 +303,9 @@ public Response initPreview(@Context UriInfo uriInfo,
    * @return the target file id
    * @throws RepositoryException the repository exception
    */
-  protected String getTargetFileId(String fileId, String workspace) throws RepositoryException {
-    ExtendedSession systemSession = (ExtendedSession) repositoryService.getCurrentRepository().getSystemSession(workspace);
-    Node node = systemSession.getNodeByIdentifier(fileId);
+    protected String getTargetFileId(String fileId, String workspace) throws RepositoryException {
+    Session systemSession =  repositoryService.getCurrentRepository().getSystemSession(workspace);
+    Node node = systemSession.getNodeByUUID(fileId);
```

Comment:
normally you need this change otherwise you will get the non referenceable node exception when previewing the symlink directly by the link ?